### ISSUE TITLE
Requeueing a job without QueueAttribute forgets entered queue 

### ISSUE THREAD

**Issue by janpieterz** [HangfireIO/Hangfire#406]
When a job that doesn't have a QueueAttribute on its handler gets re queued, it is placed in the default queue instead of its originator queue. 

I use the following code to add a job:
```
foreach (var client in clients)
{
    BackgroundJobClient.Create<IManageClientUsers>(x => x.AddUser(user), new EnqueuedState(client.QueueId));
}

This is probably since it's a little bit sneaky to work around the specific queue problem, but I'd preferably not put attributes on every handler that I'd like to be routed (in my case I'd like everything to be routed to specific queues).

Possibly related to the solution of #228

**Comment by odinserj** [comment]
That button places a job to the `default` queue, but `QueueAttribute` is able to change it. I think we should add a dropdown to the dashboard ui to choose what queue to use. Would you like to implement this feature (I can detail it)?

**Comment by janpieterz** [comment]
Yeah, if you could guide me a little bit I'm more than happy to take it on.
Would you be open for saving the originator queue so that you don't necessarily need to select a dropdown? In case the 'farm' grows bigger and certain pieces fail, I can see the need to requeue a lot of messages not to a specific queue, but to the queues they came from/are supposed to go in.

More than willing to take either or both options on, if you could guide me which files/parts I should be digging in!

**Comment by mhov** [comment]
@janpieterz you beat me to it.  I was having this same problem since I dynamically decide the queue names for jobs and ignore the "default" queue all together.  @odinserj , the Queue dropdown would be nice, but can we start off with honoring the original queue property so it makes it into the next release?

**Comment by gandarez** [comment]
I fixed the problem coding a few.

Please refer to my branch:
https://github.com/gandarez/Hangfire/tree/requeue-to-initial-queue
